migration-strategies/piercing/app/fragments/login/src/root.tsx

```diff
@@ -3,123 +3,95 @@ import {
   component$,
   useRef,
   useStore,
-  useStyles$,
   useStylesScoped$,
 } from "@builder.io/qwik";
 import { dispatchPiercingEvent } from "piercing-library";
 
 import styles from "./root.css?inline";
 
-export const Root = component$(() => {
-  const possibleInputErrors = {
-    noSpaces: "noSpaces",
-    noSpecialCharacters: "noSpecialCharacters",
-    maxLength: "maxLength",
-    emptyInput: "emptyInput",
-  } as const;
+const enum InputErrorMessages {
+  noSpaces = "Spaces are not allowed",
+  noSpecialCharacters = "Special characters are not allowed",
+  maxLength = "The provided username is too long, it needs not to be longer than 20 characters",
+  emptyInput = "Please provide a username",
+}
+interface LoginState {
```

Comment:
I think that free standing interfaces are quite nice since we can then add jsdoc comments to the members more easily than when it is inline.

Reply:
yeah good point 🙂